File: Programming/02. CSharp Part 2/07.StringsTextProcessing/02.ReverseString/ReverseString.cs
//Write a program that reads a string, reverses it and prints the result at the console.
//Example: "sample" -> "elpmas".

using System;
using System.Text;

class ReverseString
{
    static void Main()
    {
        Console.Write("Enter the string that will be reversed: ");
        string normalString = Console.ReadLine();

        StringBuilder reversedString = new StringBuilder();

        for (int letterPosition = normalString.Length - 1; letterPosition >= 0; letterPosition--)
        {
            reversedString.Append(normalString[letterPosition]);
        }

        Console.WriteLine(reversedString.ToString());
    }
}

File: HTML and CSS/JS/01.JS Fundamentals/11.Strings/js/reverseString.js
// 1 Write a JavaScript function reverses string and returns itExample: "sample" -> "elpmas".

function onReverseString() {
    printInElement('', true);
    var string = "the sample";

    printInElement("String - " + string + "\n");
    printInElement("Reversed - " + reverseString(string));


}

function reverseString(givenString) {
    var result = "";
    var stringObj = new String(givenString);
    for (var i = stringObj.length-1 ; i >= 0; i--) {
        result += stringObj[i];
    }

    return result;
}